data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -0,0 +1,498 @@
+    :param pids_dataset_id: deactivated ehr pids table in bq's dataset_id
+    :param pids_table: deactivated pids table in bq's table name
+    :return: list of queries to run
+    """
+    queries_list = []
+    dataset_list = set()
+    # Hit bq and receive df of deactivated ehr pids and deactivated date
+    deactivated_ehr_pids_df = bq.query(q=DEACTIVATED_PIDS_QUERY.render(
+        project=pids_project_id, dataset=pids_dataset_id, table=pids_table),
+                                       project_id=project_id)
+
+    date_columns_df = get_date_info_for_pids_tables(project_id)
+    LOGGER.info(
+        "Dataframe creation complete. DF to be used for creation of retraction queries."
+    )
+
+    LOGGER.info(
+        "Looping through the deactivated PIDS df to create queries based on the retractions needed per PID table"
+    )
+    for ehr_row in deactivated_ehr_pids_df.itertuples(index=False):
+        for date_row in date_columns_df.itertuples(index=False):
+            # Determine if dataset is deid to correctly pull pid or research_id and check if ID exists in dataset or if
+            # already retracted
+            if re.match(DEID_REGEX, date_row.dataset_id):
+                pid = get_research_id(date_row.project_id, date_row.dataset_id,
+                                      ehr_row.person_id)
+            else:
+                pid = ehr_row.person_id
+
+            # Check if PID is in table
+            if pid is not None and check_pid_exist(pid, date_row,
+                                                   project_id) > 0:
+                dataset_list.add(date_row.dataset_id)
+                # Get or create sandbox dataset
+                sandbox_dataset = check_and_create_sandbox_dataset(
+                    date_row.project_id, date_row.dataset_id)
+
+                # Create queries based on if the date field is null, if True, will create query based on end_date/start_date
+                if not pd.isnull(date_row.date_column):
+                    sandbox_query = SANDBOX_QUERY_DATE.render(
+                        project=date_row.project_id,
+                        sandbox_dataset=sandbox_dataset,
+                        intermediary_table=ticket_number + '_' + date_row.table,
+                        dataset=date_row.dataset_id,
+                        table=date_row.table,
+                        pid=pid,
+                        deactivated_pids_project=pids_project_id,
+                        deactivated_pids_dataset=pids_dataset_id,
+                        deactivated_pids_table=pids_table,
+                        date_column=date_row.date_column)
+                    queries_list.append({
+                        clean_consts.QUERY: sandbox_query,
+                        clean_consts.DESTINATION_DATASET: sandbox_dataset,
+                        clean_consts.DESTINATION_TABLE: date_row.table,
+                        'type': 'sandbox'
+                    })
```

Comment:
You can shorten this section by only setting the `sandbox_query` and `clean_query` in this conditional logid and moving the `queries_list.append` logic outside of the `if/else` logic.  Basically, lines 330-335 are the same as lines 365-370 and lines 345-351 are the same as 381-387.

Reply:
good catch, thank you.